corelog/logmap.go

```diff
@@ -0,0 +1,188 @@
+package corelog
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+// LogLevel is the set of all log levels.
+type LogLevel int8
+
+const (
+	// CRITICAL is the lowest log level; only errors which will end the program will be propogated.
+	CRITICAL LogLevel = -1
+	// ERROR is for errors that are not fatal but lead to troubling behavior.
+	ERROR = 0
+	// WARNING is for errors which are not fatal and not errors, but are unusual. Often sourced from misconfigurations.
+	WARNING = 1
+	// INFO is a log level for common, everyday log updates.
+	INFO = 2
+	// DEBUG is the default hidden level for more verbose updates about internal processes.
+	DEBUG = 3
+	// VERBOSE is for (potentially) call by call tracing of programs.
+	VERBOSE = 4
+)
```

Comment:
Why does the enum start at -1?

For consistency with the outside world I'd suggest use the syslog numbers:
```
           Numerical         Severity
             Code

              0       Emergency: system is unusable
              1       Alert: action must be taken immediately
              2       Critical: critical conditions
              3       Error: error conditions
              4       Warning: warning conditions
              5       Notice: normal but significant condition
              6       Informational: informational messages
              7       Debug: debug-level messages
```

Not critical to do so but consistency with other software makes sysadmins happy. ;-)